### ISSUE TITLE
management: Error when processing PermissionOffer(id='%s...'): %s

### ISSUE THREAD

**Issue by shenjiayu** [realm/realm-js#1264]
## Goals

Issue tokens from permission offer

## Expected Results

Get tokens back from realm object server

## Actual Results
The following error occurred on the server side
`management: Error when processing PermissionOffer(id='%s...'): %s`

## Code Sample
```
constructor() {
    super();

    this.state = {
      token: null
    };

    let user = Realm.Sync.User.current;
    if (user) {
      this.managementRealm = user.openManagementRealm();
  
      if (!this.offers) {
        this.offers = this.managementRealm.objects('PermissionOffer');
        this.offers.addListener((offers, changes) => {
          if (offers.length > 0) {
            this.setState({
              token: offers[0].token
            });
            console.log(offers[0]);
          }
        });
      }
    } else {
      Realm.Sync.User.login(url, email, password, (error, user) => {
        if (error) {
          console.log(error);
        } else {
          
        }
      });
    }

    this.requestToken = this.requestToken.bind(this);
  }

  requestToken() {
    this.managementRealm.write(() => {
      var expirationDate = new Date();
      expirationDate.setDate(expirationDate.getDate() + 7);
      this.managementRealm.create('PermissionOffer', {
        id: uuidv4(),
        createdAt: new Date(),
        updatedAt: new Date(),
        userId: <userId>,
        realmUrl: '*',
        mayRead: true,
        mayWrite: true,
        mayManage: false,
        expiresAt: expirationDate
      });
    });
  }
```
## Version of Realm and Tooling

- Realm JS SDK Version: 1.10.3
- Node or React Nattive: 0.47.2
- Client OS & Version: OSX 10.12.6
- Which debugger for React Native: None

**Comment by kristiandupont** [comment]
Hi @shenjiayu, 
if you store the `PermissionOffer` object:
```javascript
const permissionOffer = this.managementRealm.create('PermissionOffer', { /* ... */ });
```

you can inspect the `permissionOffer.statusMessage` property to get a descriptive error message.
Note that we have just merged at number of helper methods for permission handling, so this should be simplified quite a bit. It will be included in the next release which we will hopefully get around to this week.

**Comment by shenjiayu** [comment]
@kristiandupont the `statusMessage` is null then, and the back-end server is still popping `management: Error when processing PermissionOffer(id='%s...'): %s` error.

**Comment by kristiandupont** [comment]
I understand why you would make that assumption but it's not entirely correct. When `write()` completes, you are guaranteed that the object is written in your local database. It will subsequently be synchronized with the server. Then server will *then* process the request, update the object which will then be synchronized with your local realm again. So you should wait a little bit before checking the value. Unfortunately, we do not yet support object-level notifications in the JS sdk, but you can see our workaround for this very issue here: https://github.com/realm/realm-js/blob/master/lib/permission-api.js#L105

**Comment by shenjiayu** [comment]
@kristiandupont Thx for your help. I will use `permissionChange` for the moment. 
Glad to see you guys put a lot effort to bring js client onto the right track.

**Comment by kneth** [comment]
@shenjiayu We have released 1.11.0 with our new permission management methods. The documentation at the website isn't published yet but will be shortly.
